src/client/executor.rs

```diff
@@ -99,7 +99,7 @@ impl Client {
         op: T,
         session: impl Into<Option<&mut ClientSession>>,
     ) -> Result<ExecutionDetails<T>> {
-        Box::pin(async {
```

Comment:
Thanks for this explanation!
I wonder if this can bring performance gain if we reduce the size of futures?
AFAIK, using Box here will bring runtime const and make the compiler unable to inline this future.